services/terrain/src/terrain/services/permanent_id_requests.clj

```diff
@@ -38,11 +44,21 @@
              :path staging-dest})))
   data-item)
 
+(defn- validate-publish-dest
```

Comment:
I notice that both this and `validate-staging-dest` are checking existence, so it might save a bit to only call `existence-marker` once.